semgrep-core/src/analyzing/AST_to_IL.ml

```diff
@@ -386,6 +427,33 @@ and expr_aux env eorig =
           add_instr env (mk_i (Assign (lval, opexp)) eorig);
           lvalexp
       | _ -> impossible (G.E eorig))
+  | G.Call
```

Comment:
Curious why you took a different approach here than in the previous. It seems like it could also be reasonable to assume `$X.$Y(...);` mutates `$X` (which I think you do later in dataflow?)

Reply:
Very good observation/question! In the case of arbitrary void-method calls we just assume that "anything can happen", so we the callee may end up with some arbitrary non-constant value. But `concat` is something we know about, so when we find `a.concat("hello")` we want to know that if `a` was constant it continues to be. Maybe I should add a test for this case too.

Reply:
Added a test for such case.